src/services/proofGenerationServices.js

```diff
@@ -142,46 +135,39 @@ export async function generateExitPayload(burnTxHash, eventSignature, isMainnet)
   const rpcLength = isMainnet ? mainnetRPCLength : testnetRPCLength
   const initialRpcIndex = isMainnet ? config.mainnetRpcIndex : config.testnetRpcIndex
 
-  var matic
-  var maticPoS
   var result
 
   // loop over rpcs to retry in case of an in case of an rpc error
   for (var i = 0; i < maxRetries; i++) {
     const rpcIndex = (initialRpcIndex + i) % rpcLength
     try {
       // initialize matic client
-      await initMatic(isMainnet, maticRPC[rpcIndex], ethereumRPC[rpcIndex]).then((obj) => {
-        matic = obj.matic
-        maticPoS = obj.maticPoS
-      })
+      const maticClient = await initMatic(isMainnet, maticRPC[rpcIndex], ethereumRPC[rpcIndex])
 
       // fetch last child block included
-      const lastChildBlock = await matic.rootChain.getLastChildBlock()
+      const lastChildBlock = await maticClient.exitUtil.rootChain.getLastChildBlock()
 
       // fetch transaction receipt
-      const receipt = await matic.web3Client.getMaticWeb3().eth.getTransactionReceipt(burnTxHash)
-
-      if (!receipt || !receipt.blockNumber) {
-      // temporary fix for when we receive null receipts
-      // should be reverted back when issue is fixed
+      try {
+        const receipt = await maticClient.client.child.getTransactionReceipt(burnTxHash)
```

Comment:
here you can use `isCheckPointed` API.https://maticnetwork.github.io/matic.js/docs/pos/is-check-pointed/

Reply:
Needed to differentiate between the receiving null receipt for a correct txHash and trying to fetch a wrong txHash to implement the retry logic.